ros2_batch_job/packaging.py

```diff
@@ -112,30 +112,31 @@ def build_and_test_and_package(args, job):
     # Only on Linux and OSX Python scripts have a shebang line
     if args.os in ['linux', 'osx']:
         print('# BEGIN SUBSECTION: rewrite shebang lines')
-        bin_path = os.path.join(args.installspace, 'bin')
-        for filename in os.listdir(bin_path):
-            path = os.path.join(bin_path, filename)
-            with open(path, 'rb') as h:
-                content = h.read()
-            shebang = b'#!%b' % job.python.encode()
-            if content[0:len(shebang)] != shebang:
-                continue
-            print('- %s' % path)
-            with open(path, 'wb') as h:
-                h.write(b'#!/usr/bin/env python3')
-                h.write(content[len(shebang):])
+        # Demo nodes are installed to 'lib', so binaries may be present there in addition to 'bin'
+        for dir_with_binaries in ['bin', 'lib']:
+            path_with_binaries = os.path.join(args.installspace, dir_with_binaries)
+            for path in [os.path.join(root, f) for root, dirs, files in os.walk(path_with_binaries) for f in files]:
```

Comment:
that's what I am trying to understand. If in the future we install something else that has `#!/Users/osrf/jenkins/workspace/packaging_osx/venv/bin/python` at the top of it (that's all that gets matched), why we would ever want to leave it? I'll make it behave differently for the two directories to be conservative as you suggest, but it sounds like you are aware of a situation that I'm not so I am trying to pick your brain :smile: